libs/clients/zebpay/client.go

```diff
@@ -248,25 +253,33 @@ func affixAccessToken(req *http.Request, opts *ClientOpts) error {
 
 // CheckTransfer uploads the bulk payout for gemini
 func (c *HTTPClient) CheckTransfer(ctx context.Context, opts *ClientOpts, id uuid.UUID) (*CheckTransferResponse, error) {
+	var resp = new(CheckTransferResponse)
+	resp.ID = id
 	if opts == nil {
-		return nil, errBadClientOpts
+		resp.Error = errBadClientOpts.Error()
+		return resp, errBadClientOpts
 	}
 
 	// /api/checktransferstatus/<UUID>/status
 
 	urlPath := fmt.Sprintf("/api/checktransferstatus/%s/status", id.String())
 	req, err := c.client.NewRequest(ctx, "GET", urlPath, nil, nil)
 	if err != nil {
-		return nil, err
+		resp.Error = err.Error()
+		return resp, err
 	}
 
 	// populate the access token
 	affixAccessToken(req, opts)
 
-	var resp = new(CheckTransferResponse)
 	_, err = c.client.Do(ctx, req, resp)
 	if err != nil {
-		return nil, fmt.Errorf("error %v encountered when checking status. response: %v", err, resp)
+		resp.Error = err.Error()
+		if errorBundle, ok := err.(*errorutils.ErrorBundle); ok {
+			if errorState, ok := errorBundle.Data().(clients.HTTPState); ok {
+				resp.Code = int64(errorState.Status)
+			}
+		}
```

Comment:
This won't work for wrapped errors, which these days is a likely scenario. Got to use `errors.As`:

```suggestion
		var eerb *errorutils.ErrorBundle
		if errors.As(err, &errb) {
			if state, ok := errb.Data().(clients.HTTPState); ok {
				resp.Code = int64(state.Status)
			}
		}
```